### ISSUE TITLE
Invalide PythonVersionIndex (C:\Delphi\progs32\Python\PyScripter\Source

### ISSUE THREAD

**Issue by ManPython** [pyscripter/pyscripter#877]
Exception class: EAssertionFailed
Exception message: Invalide PythonVersionIndex (C:\Delphi\progs32\Python\PyScripter\Source\cPyControl.pas, line 293).
Exception address: 00A37850
------------------------------------------------------------------------------
PyScripter version : 3.4.0.0 x86

**Comment by pyscripter** [comment]
Could you please provide some more info.   I want to fix this asap.   

Do you get an Exception dialog with call stack information?  Can you send the report by email or copy and paste its content here?

Can you describe the steps that led to this error?

**Comment by ManPython** [comment]
1. I just downloaded .zip version of x86 and run - then 1st post of bug as window dialog 
a) now I discover the dialog
![dialog 1](https://user-images.githubusercontent.com/25403861/39678241-266f0906-5189-11e8-9568-6320ca9606ec.png)
b) after ok
![dialogs](https://user-images.githubusercontent.com/25403861/39678256-4802f956-5189-11e8-8910-2f065771c855.png)

2. After no any python discovered in Pyscript, I planed add manualy - then 2st post of bug as window (I was in idea that some code calling your enviorment as portable version)

3. Then I plan install app, and discover that i downloaded x86 but my system i x64, the x64 from .zip working coretlly without instalation (discover pythons, etc.)

My pythons are for x64, but I planed test x86

**Comment by ManPython** [comment]
I noticed that icons are missed in x86

![pyscripter](https://user-images.githubusercontent.com/25403861/39678287-c5da7d5e-5189-11e8-9598-e9be1233063f.png)

**Comment by pyscripter** [comment]
If you still have the 32bit version would it be possible to press the "Details" button and post the contents here?

**Comment by ManPython** [comment]
@pyscripter the post 1,2 are as data from details

**Comment by pyscripter** [comment]
No icons are not missed.  Because no Python is loaded the corresponding actions are disabled.   In the Windows 10 Dark theme, disabled actions do not show up.

**Comment by pyscripter** [comment]
Yes but I would like to see the full lot if possible.

**Comment by ManPython** [comment]
Backing to point 2, I try to "setup python" but no reaction.

**Comment by ManPython** [comment]
I thing that logs are full enough, here you are ; )
![full lot](https://user-images.githubusercontent.com/25403861/39678327-bc6724ce-518a-11e8-9870-f3f29047c09b.png)

**Comment by pyscripter** [comment]
So you get no call stack information.

And the steps that produced the exception were the following:
-  you run Pyscripter 32bit on Windows 64 with no 32bit python installed.
- Pressed OK
- The Exception occurred.

**Comment by ManPython** [comment]
Thats correct. I think that Pys32bit should work normaly on Wx64 and in other variations of configuration.
Maybe  Pys64bit need check behaviors with Pythons x86 as experiment.

**Comment by pyscripter** [comment]
Yes it should.  I will upload an update later tonight fixing the issue.

**Comment by pyscripter** [comment]
I did reproduce the error and will fix it.   However I was able to go into the Python Versions dialog and carrying on with configuring python versions.   Please note that the 32bit version of Pyscripter will only recognize and work with 32bit versions of Python and the 64bit version of Pyscripter will only recognize and work with 64bit versions of Python.   On 64bit Windows you can use either or both.

**Comment by ManPython** [comment]
> However I was able to go into the Python Versions dialog and carrying on with configuring python versions
When I try to select path then no reaction. No any python discovered or somithing.
Btw.. the idea of clicking catalogs are not friendly. Should exist the wayt to put prepared path in some input field. In case of many venv etc.. clicking can help to lose patention ; )

I think that release vesrions should be available from github or alternative from current, and use to autoupdate.

**Comment by pyscripter** [comment]
I missed the bit about the Python Path in your bug report.

The Python Path command should not be available if Python is not laoded.   (This is fixed in version control but not in 3.4.1).

The way to add unregistered python versions (or otherwise unrecognized versions) to Pyscripter is through Python Setup... and in the dialog box by pressing the Add (+) button.
